editor/components/inserter/menu.js

```diff
@@ -12,11 +13,58 @@ import Dashicon from 'components/dashicon';
 class InserterMenu extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
+		this.blockTypes = wp.blocks.getBlocks();
+		this.categories = wp.blocks.getCategories();
+		this.nodes = {};
 		this.state = {
-			filterValue: ''
+			filterValue: '',
+			currentFocus: null
 		};
 		this.filter = this.filter.bind( this );
 		this.instanceId = this.constructor.instances++;
+		this.isShownBlock = this.isShownBlock.bind( this );
+		this.setSearchFocus = this.setSearchFocus.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.getVisibleBlocks = this.getVisibleBlocks.bind( this );
+		this.sortBlocksByCategory = this.sortBlocksByCategory.bind( this );
+	}
+
+	componentDidMount() {
+		document.addEventListener( 'keydown', this.onKeyDown );
+	}
+
+	componentWillUnmount() {
+		document.removeEventListener( 'keydown', this.onKeyDown );
+	}
+
+	isShownBlock( block ) {
+		return block.title.toLowerCase().indexOf( this.state.filterValue.toLowerCase() ) !== -1;
+	}
+
+	bindReferenceNode( nodeName ) {
+		return ( node ) => this.nodes[ nodeName ] = node;
+	}
+
+	isNextKeydown( keydown ) {
+		return keydown.code === 'ArrowDown'
```

Comment:
Ah maybe that is the source of some of the a11y problems.  Un-normalized events are no fun.